### PR TITLE
OAuth ROPC flow should prefer user security name if different than supplied name

### PR DESCRIPTION
**Pull request by brutif**

### CODE REVIEW

**Comment by brutif** [comment]
#spawn.full.fats=com.ibm.ws.security.oauth_fat,com.ibm.ws.security.openidconnect.server_fat,com.ibm.ws.security.openidconnect.server_fat.config,com.ibm.ws.security.openidconnect.server_fat.endpoint.authorize,com.ibm.ws.security.openidconnect.server_fat.endpoint.clientregistration,com.ibm.ws.security.openidconnect.server_fat.endpoint.coveragemap,com.ibm.ws.security.openidconnect.server_fat.endpoint.discovery,com.ibm.ws.security.openidconnect.server_fat.endpoint.end_session,com.ibm.ws.security.openidconnect.server_fat.endpoint.introspect,com.ibm.ws.security.openidconnect.server_fat.endpoint.revoke,com.ibm.ws.security.openidconnect.server_fat.endpoint.token,com.ibm.ws.security.openidconnect.server_fat.endpoint.userinfo,com.ibm.ws.security.openidconnect.server_fat.feature,com.ibm.ws.security.openidconnect.server_fat.granttype.jwt,com.ibm.ws.security.openidconnect.server_fat.gui.tokenMgmt,com.ibm.ws.security.openidconnect.server_fat.jaxrs,com.ibm.ws.security.openidconnect.server_fat.jaxrs.config,com.ibm.ws.security.openidconnect.server_fat.oauth,com.ibm.ws.security.openidconnect.server_fat.oidc,com.ibm.ws.security.openidconnect.server_fat.saml,com.ibm.ws.security.openidconnect.server_fat.spnego

File: dev/com.ibm.ws.security.oauth/resources/OSGI-INF/l10n/metatype.properties [hunk 1]
```diff
@@ -344,6 +344,9 @@ internalClientId.desc=The ID of an existing client that will be used to generate
 internalClientSecret=Client secret used to generate tokens
 internalClientSecret.desc=The secret of an existing client that will be used to generate tokens for PersonalTokenManagement and UsersTokenManagement UI pages.
 
+ropcPreferUserSecurityName=Resource Owner Password Credentials Grant (ROPC) Prefers User Security Name
+ropcPreferUserSecurityName.desc=Boolean indicating that the user name in tokens created by the ROPC Grant type will be set to the User Security Name if it differs from the user name supplied in the ROPC request. The default is false.
```
Comment (Charlotte-Holt) [inline dev/com.ibm.ws.security.oauth/resources/OSGI-INF/l10n/metatype.properties]: How does this sound for the description? Is it still technically correct? 
"If the user security name differs from the username that is supplied in the ROPC request, then the username is set to the user security name for all tokens that are created by the ROPC grant type. The default is false."

File: dev/com.ibm.ws.security.oauth/resources/OSGI-INF/l10n/metatype.properties [hunk 2]
```diff
@@ -344,6 +344,9 @@ internalClientId.desc=The ID of an existing client that will be used to generate
 internalClientSecret=Client secret used to generate tokens
 internalClientSecret.desc=The secret of an existing client that will be used to generate tokens for PersonalTokenManagement and UsersTokenManagement UI pages.
 
+ropcPreferUserSecurityName=Resource Owner Password Credentials Grant (ROPC) Prefers User Security Name
```
Comment (Charlotte-Holt) [inline dev/com.ibm.ws.security.oauth/resources/OSGI-INF/l10n/metatype.properties]: Change to:
"Resource Owner Password Credentials (ROPC) grant prefers user security name"

Comment (brutif) [reply dev/com.ibm.ws.security.oauth/resources/OSGI-INF/l10n/metatype.properties]: done

File: dev/com.ibm.ws.security.oauth/resources/OSGI-INF/l10n/metatype.properties (hunk 1 above)
Comment (brutif) [reply dev/com.ibm.ws.security.oauth/resources/OSGI-INF/l10n/metatype.properties]: That'll work.  Done.

**Comment by brutif** [comment]
#build
#spawn.fullfat.buckets=com.ibm.ws.security.oauth_fat,com.ibm.ws.security.openidconnect.server_fat,com.ibm.ws.security.openidconnect.server_fat.config,com.ibm.ws.security.openidconnect.server_fat.endpoint.authorize,com.ibm.ws.security.openidconnect.server_fat.endpoint.clientregistration,com.ibm.ws.security.openidconnect.server_fat.endpoint.coveragemap,com.ibm.ws.security.openidconnect.server_fat.endpoint.discovery,com.ibm.ws.security.openidconnect.server_fat.endpoint.end_session,com.ibm.ws.security.openidconnect.server_fat.endpoint.introspect,com.ibm.ws.security.openidconnect.server_fat.endpoint.revoke,com.ibm.ws.security.openidconnect.server_fat.endpoint.token,com.ibm.ws.security.openidconnect.server_fat.endpoint.userinfo,com.ibm.ws.security.openidconnect.server_fat.feature,com.ibm.ws.security.openidconnect.server_fat.granttype.jwt,com.ibm.ws.security.openidconnect.server_fat.gui.tokenMgmt,com.ibm.ws.security.openidconnect.server_fat.jaxrs,com.ibm.ws.security.openidconnect.server_fat.jaxrs.config,com.ibm.ws.security.openidconnect.server_fat.oauth,com.ibm.ws.security.openidconnect.server_fat.oidc,com.ibm.ws.security.openidconnect.server_fat.saml,com.ibm.ws.security.openidconnect.server_fat.spnego

**Comment by brutif** [comment]
#run-libby-bot

**Comment by brutif** [comment]
pb looked good except for schema val. failure, fixed that.
I think the UI fails were random, need to be sure they don't appear in the next build.
#run-libby-bot
#build

**Pull request merged**